src/Jellyfin.XmlTv/XmlTvReader.cs

```diff
@@ -1021,49 +1020,61 @@ private void PopulateHeader(XmlReader reader, XmlTvProgram result)
                 return null;
             }
 
-            var completeDate = "20000101000000";
+            const string CompleteDate = "20000101000000";
             var dateComponent = string.Empty;
-            var dateOffset = "+00:00";
+            string? dateOffset = null;
             var match = Regex.Match(dateValue, DateWithOffsetRegex);
             if (match.Success)
             {
                 dateComponent = match.Groups["dateDigits"].Value;
                 if (!string.IsNullOrEmpty(match.Groups["dateOffset"].Value))
                 {
-                    dateOffset = match.Groups["dateOffset"].Value; // Add in the colon to ease parsing later
-                    if (dateOffset.Length == 5)
+                    var tmpDateOffset = match.Groups["dateOffset"].Value; // Add in the colon to ease parsing later
+                    if (tmpDateOffset.Length == 5)
                     {
-                        dateOffset = dateOffset.Insert(3, ":"); // Add in the colon to ease parsing later
-                    }
-                    else
-                    {
-                        dateOffset = "+00:00";
+                        dateOffset = tmpDateOffset.Insert(3, ":"); // Add in the colon to ease parsing later
                     }
                 }
             }
+            else
+            {
+                return null;
+            }
 
             // Pad out the date component part to 14 characaters so 2016061509 becomes 20160615090000
             if (dateComponent.Length < 14)
             {
-                dateComponent = dateComponent + completeDate.Substring(dateComponent.Length, completeDate.Length - dateComponent.Length);
+                dateComponent += CompleteDate[dateComponent.Length..];
             }
 
-            var standardDate = string.Format(
-                CultureInfo.InvariantCulture,
-                "{0} {1}",
-                dateComponent,
-                dateOffset);
-            if (DateTimeOffset.TryParseExact(standardDate, "yyyyMMddHHmmss zzz", CultureInfo.CurrentCulture, DateTimeStyles.None, out DateTimeOffset parsedDateTime))
+            if (dateOffset == null)
             {
-                return parsedDateTime.ToUniversalTime();
+                if (DateTimeOffset.TryParseExact(dateComponent, "yyyyMMddHHmmss", CultureInfo.CurrentCulture, DateTimeStyles.AssumeUniversal, out DateTimeOffset parsedDateTime))
+                {
+                    return parsedDateTime;
+                }
+
+                return null;
```

Comment:
Shouldn't this just be outside the if block seeing as both branches return null?